src/test/java/org/openrewrite/java/migrate/guava/NoGuavaSetsFilterTest.java

```diff
@@ -0,0 +1,143 @@
+    public void defaults(RecipeSpec spec) {
+        spec
+          .recipe(new NoGuavaSetsFilter())
+          .parser(JavaParser.fromJavaVersion().classpathFromResources(new InMemoryExecutionContext(), "guava"));
+    }
+
+    @DocumentExample
+    @Test
+    void replaceSetsFilter() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import java.util.Set;
+
+              import com.google.common.base.Predicate;
+              import com.google.common.collect.Sets;
+
+              class Test {
+                  public static Set<Object> test(Set<Object> set, Predicate<Object> isNotNull) {
+                      return Sets.filter(set, isNotNull);
+                  }
+              }
+              """,
+            """
+              import java.util.Set;
+              import java.util.stream.Collectors;
+
+              import com.google.common.base.Predicate;
+
+              class Test {
+                  public static Set<Object> test(Set<Object> set, Predicate<Object> isNotNull) {
+                      return set.stream().filter(isNotNull).collect(Collectors.toSet());
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void replaceSetsFilterUsingSortedSet() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import java.util.Set;
+              import java.util.SortedSet;
+
+              import com.google.common.base.Predicate;
+              import com.google.common.collect.Sets;
+
+              class Test {
+                  public static Set<Object> test(SortedSet<Object> set, Predicate<Object> isNotNull) {
+                      return Sets.filter(set, isNotNull);
+                  }
+              }
+              """,
+            """
+              import java.util.Set;
+              import java.util.SortedSet;
+              import java.util.TreeSet;
+              import java.util.stream.Collectors;
+
+              import com.google.common.base.Predicate;
+
+              class Test {
+                  public static Set<Object> test(SortedSet<Object> set, Predicate<Object> isNotNull) {
+                      return set.stream().filter(isNotNull).collect(Collectors.toCollection(TreeSet::new));
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void replaceSetsFilterIndirectlyUsingSortedSet() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import java.util.Set;
+              import java.util.SortedSet;
+
+              import com.google.common.base.Predicate;
+              import com.google.common.collect.Sets;
+
+              class Test {
+                  public static Set<Object> test(SortedSet<Object> set, Predicate<Object> isNotNull) {
+                      Set<Object> indirectSet = set;
+                      return Sets.filter(indirectSet, isNotNull);
```

Comment:
This case is tricky; we can't really deduce from the type system whether we'd end up in this conditional of special handling:
https://github.com/google/guava/blob/21cdcf6d9a49c1a498acb422471d8d71f9219865/guava/src/com/google/common/collect/Sets.java#L1190-L1194

Reply:
I'd hope that to be rare enough to perhaps still merge the current work, but there is a risk here which could only be mitigated by data flow analysis to figure out if it might be a SortedSet going into this `filter(Set, Predicate)` method. 🤔

Reply:
Either way thanks for the quick rework; would love to hear your thoughts on this edge case.